lib/Sema/CSSimplify.cpp

```diff
@@ -4846,6 +4846,19 @@ ConstraintSystem::simplifyKeyPathConstraint(Type keyPathTy,
       return SolutionKind::Solved;
     };
 
+  // We do not allow KeyPaths to go through AnyObject, so let's create a fix
+  // and allow that. This will get diagnosed later.
+  if (auto fixedRootTy =
+          getFixedTypeRecursive(rootTy, subflags, /*wantRValue=*/true)) {
+    if (fixedRootTy->isAnyObject()) {
```

Comment:
if fixedRootTy is a type variable, it might be assigned a type of AnyObject later. Perhaps you should add `if (fixedRootTy->isTypeVariableOrMember()) return;` here?

Reply:
I can't return `SolutionKind::Solved` before this check otherwise it leads to a crash in expression rewriter.

Reply:
The checks are mutually exclusive. You want to return Unsolved if it’s a type variable or member; see how this is handled for other constraint kinds such as ConformsTo.

Reply:
Oh, well no matter what I return, it leads to a crash. If I return Unsolved, I hit an unreachable: `should have generated constraints`.

Reply:
You’ll need to re-add the same key path constraint; since it’s unsolved, it has to be solved again once one of the type variables is bound to a type.

Reply:
Oh, thank you! I have added an unsolved constraint :)